### PR TITLE
fix clean target from build Makefile

### PR DESCRIPTION
**Pull request by JDLopes**

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -107,7 +107,22 @@ test: $(TEST_TYPE_LIST)
 # CLEAN
 #
 
-clean: fw-clean pc-emul-clean sim-clean fpga-clean doc-clean
+clean:
+ifneq ($(wildcard $(EMB_DIR)/.),)
```
Comment (jjts) [inline build/Makefile:111]: please let the clean target as it was before and move the ifneq tests to the respective targets fw-clean pc-emul

-- you save 5 lines in this way ;-)

Comment (JDLopes) [reply build/Makefile]: Done! :-)

File: build/Makefile [hunk 2]
```diff
@@ -88,9 +96,6 @@ doc-test:
 
 doc-debug: 
 	make -C $(DOC_DIR) debug
-
-doc-clean: 
-	make -C $(DOC_DIR) clean
 endif
```
Comment (jjts) [inline build/Makefile:99]: use TEST_TYPE_LIST

use TEST_TYPE_LIST+=

$(TEST_LIST) in PHONY list and remove *-test

File: build/Makefile [hunk 3]
```diff
@@ -26,16 +31,17 @@ pc-emul-run:
...
-	make -C $(PC_DIR) clean
 endif
 
 #
 # SIMULATE
 #
 SIM_DIR=hw/sim
+
+sim-clean:
 ifneq ($(wildcard $(SIM_DIR)/.),)
```
Comment (jjts) [inline build/Makefile:42]: simulation is mandatory no need to test

File: build/Makefile (hunk 2 above)
Comment (JDLopes) [reply build/Makefile:98]: Done!

File: build/Makefile (hunk 3 above)
Comment (JDLopes) [reply build/Makefile]: Done!

**Pull request merged**